### DISCUSSION TITLE
Is value set with CURLOPT_USERAGENT overriden when using custom headers with CURLOPT_HTTPHEADER?

### DISCUSSION THREAD

**Discussion by ribtoks** [curl/curl#8118] [Q&A] [answered]
I mean when `CURLOPT_HTTPHEADER` is set with any headers. Do the "shortcuts" like `_USERAGENT`, `_REFERRER` etc. are thrown away?

It is not exactly clear from [documentation](https://curl.se/libcurl/c/CURLOPT_USERAGENT.html).

**Comment by jay** [comment, marked as answer]
In recent versions the header list overrides the set user agent and referer but I'm not sure if it's always been that way. I would set one or the other. Is there a reason you need to set both?

**Reply by ribtoks** [reply]
@jay  thanks for answering. I was setting `USER_AGENT` because it was a separate option but did not observe it in the logs of the request that was being sent so it was quite confusing.

**Reply by jay** [reply]
To clarify, if you set CURLOPT_HTTPHEADER with a User-Agent header that will override the user agent set via CURLOPT_USERAGENT. If you set CURLOPT_HTTPHEADER _without_ a User-Agent header then libcurl will use CURLOPT_USERAGENT. This behavior is undocumented and therefore subject to change. Use one or the other to set the user agent.

**Reply by ribtoks** [reply]
All right, understood. Thank you!
